File: retropreference/src/main/java/com/twiceyuan/library/typeHandler/FloatHandler.java
package com.twiceyuan.library.typeHandler;

import android.content.SharedPreferences;

/**
 * Created by twiceYuan on 20/01/2017.
 *
 * Float handler
 */
public class FloatHandler extends BaseTypeHandler<Float> {

    public FloatHandler(SharedPreferences preferences) {
        super(preferences);
    }

    @Override
    public void setValue(String key, Float value) {
        mPreferences.edit().putFloat(key, value).apply();
    }

    @Override
    public Float get(String key, Float defaultValue) {
        return mPreferences.getFloat(key, defaultValue);
    }

    @Override
    public Float defaultValue() {
        return 0f;
    }
}

File: retropreference/src/main/java/com/twiceyuan/retropreference/typeHandler/FloatHandler.kt
package com.twiceyuan.retropreference.typeHandler

import android.content.SharedPreferences

/**
 * Created by twiceYuan on 20/01/2017.

 * Float handler
 */
internal class FloatHandler(preferences: SharedPreferences) : BaseTypeHandler<Float>(preferences) {

    override fun setValue(key: String, value: Float?) {
        mPreferences.edit().putFloat(key, value ?: 0f).apply()
    }

    override operator fun get(key: String, defaultValue: Float?): Float? {
        return mPreferences.getFloat(key, defaultValue ?: 0f)
    }

    override fun defaultValue(): Float? {
        return 0f
    }
}
